src/test/java/unit/AiNotebookControlled.java

```diff
@@ -0,0 +1,312 @@
+    String name = "resolveAndCheckAccess";
+    UFAiNotebook createdNotebook =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class, "resources", "create", "ai-notebook", "--name=" + name);
+
+    // `terra resources resolve --name=$name --format=json`
+    String resolved =
+        TestCommand.runAndParseCommandExpectSuccess(
+            String.class, "resources", "resolve", "--name=" + name);
+    assertEquals(createdNotebook.instanceName, resolved, "resolve returns the instance name");
+
+    // `terra resources check-access --name=$name`
+    String stdErr =
+        TestCommand.runCommandExpectExitCode(1, "resources", "check-access", "--name=" + name);
+    assertThat(
+        "check-access error is because ai notebooks are controlled resources",
+        stdErr,
+        CoreMatchers.containsString("Checking access is intended for REFERENCED resources only"));
+  }
+
+  @Test
+  @DisplayName("override the default location and instance id")
+  void overrideLocationAndInstanceId() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources create ai-notebook --name=$name
+    // --cloning=$cloning --description=$description
+    // --location=$location --instance-id=$instanceId`
+    String name = "overrideLocationAndInstanceId";
+    CloningInstructionsEnum cloning = CloningInstructionsEnum.REFERENCE;
+    String description = "\"override default location and instance id\"";
+    String location = "us-central1-b";
+    String instanceId = "a" + UUID.randomUUID().toString(); // instance id must start with a letter
+    UFAiNotebook createdNotebook =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class,
+            "resources",
+            "create",
+            "ai-notebook",
+            "--name=" + name,
+            "--cloning=" + cloning,
+            "--description=" + description,
+            "--location=" + location,
+            "--instance-id=" + instanceId);
+
+    // check that the properties match
+    assertEquals(name, createdNotebook.name, "create output matches name");
+    assertEquals(cloning, createdNotebook.cloningInstructions, "create output matches cloning");
+    assertEquals(description, createdNotebook.description, "create output matches description");
+    assertEquals(location, createdNotebook.location, "create output matches location");
+    assertEquals(instanceId, createdNotebook.instanceId, "create output matches instance id");
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, createdNotebook.accessScope, "create output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        createdNotebook.privateUserName.toLowerCase(),
+        "create output matches private user name");
```

Comment:
optional: Unless there's a particular reason to think that changing the location and instance id will change the other fields, we could probably skip re-asserting in this test that the name, description, cloning, access, & email match and teset that once in the other test. ditto for description matches.

Reply:
Fair point. Going to keep them since I've already got them and they are testing CLI code, but agreed it's not the most critical part of this test.